pallets/market/src/lib.rs

```diff
@@ -566,6 +580,79 @@ pub mod pallet {
             Ok(())
         }
 
+        /// Publish a new set of storage deals (not yet included in a sector).
+        /// It saves valid deals as [`DealState::Published`] and locks up client fees and provider's collaterals.
+        /// Locked up balances cannot be withdrawn until a deal is terminated.
+        /// All of the deals must belong to a single Storage Provider.
+        /// It is permissive, if some of the deals are correct and some are not, it emits events for valid deals.
+        /// On success emits [`Event::<T>::DealPublished`] for each successful deal.
+        #[pallet::call_index(2)]
+        #[pallet::weight((T::WeightInfo::settle_deal_payments(deals.len() as u32), DispatchClass::Normal))]
```

Comment:
`settle_deal_payments` weight for `publish_storage_deals`?

Reply:
Good catch